CMakeLists.txt

```diff
@@ -14,6 +14,12 @@ if(POLICY CMP0135)
     cmake_policy(SET CMP0135 NEW) # valid for DOWNLOAD_EXTRACT_TIMESTAMP option in CMake 3.24 and later
 endif()
 
+# New in CMake version 3.30. FetchContent_Populate(<name>) is deprecated, call
+# FetchContent_MakeAvailable(<name>) instead.
```

Comment:
`FetchContent_MakeAvailable` was introduced in 3.14. A version check could be used to call it instead to guard against the removal of this policy.

https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable

Reply:
Thanks for the suggestion. I did not mention that I tried to do the right thing. Our code in cmake/Modules/AvifExternalProjectUtils.cmake does the following (irrelevant details omitted):

```
macro(avif_fetchcontent_populate_cmake name)
    if(NOT ${name}_POPULATED)
        FetchContent_Populate(${name})
        ...
        add_subdirectory(${${name}_SOURCE_DIR} ${${name}_BINARY_DIR} EXCLUDE_FROM_ALL)
        ...
    endif()
endmacro()
```

Note that we pass `EXCLUDE_FROM_ALL` to the `add_subdirectory()` call. To replace the code above with a `FetchContent_MakeAvailable()` call, we will need to pass `EXCLUDE_FROM_ALL` to all the `FetchContent_Declare()` calls. But passing `EXCLUDE_FROM_ALL` to `FetchContent_Declare()` is a new feature in CMake 3.28:

https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare

Therefore, the cmake version check we need is 3.28, not 3.14, and we will need to add a new function that takes care of the `FetchContent_Declare()` + `FetchContent_MakeAvailable()` sequence. At that point I concluded I should just set the `CMP0169` policy to `OLD`.

Reply:
James: We can consider requiring CMake 3.16 as the minimum version, based on the [Google Foundational C++ support table](https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md).